backend/cmd/headlamp.go

```diff
@@ -1001,59 +1019,90 @@ func (c *HeadlampConfig) getConfig(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+//nolint:funlen,nestif
 func (c *HeadlampConfig) addCluster(w http.ResponseWriter, r *http.Request) {
 	clusterReq := ClusterReq{}
 	if err := json.NewDecoder(r.Body).Decode(&clusterReq); err != nil {
-		fmt.Println(err)
 		http.Error(w, "Error decoding cluster info", http.StatusBadRequest)
-
 		return
 	}
 
-	if clusterReq.Name == "" || clusterReq.Server == "" {
+	if clusterReq.KubeConfig != nil {
+		kubeConfigByte, err := base64.StdEncoding.DecodeString(*clusterReq.KubeConfig)
+		if err != nil {
+			http.Error(w, "Error decoding kubeconfig", http.StatusBadRequest)
+			return
+		}
+
+		config, err := clientcmd.Load(kubeConfigByte)
+		if err != nil {
+			http.Error(w, "Error loading kubeconfig", http.StatusBadRequest)
+			return
+		}
+
+		err = writeKubeConfig(*config, c.kubeConfigPersistenceDir)
```

Comment:
I guess we can add this now but worry later about what to do with clusters that should be added dynamically in a multi-user env (we'll probably add the cluster info to each request itself).

Reply:
Yes, like we discussed earlier we have to figure out how to solve for dynamic multi-user env separately.